src/content/en/2020/http2.md

```diff
@@ -4,20 +4,534 @@ part_number: IV
+
+Another benefit of QUIC is that it is able to **“migrate” connections** and keep them alive even when the underlying network changes. A typical example is the so-called “parking lot problem”. Imagine your smartphone is connected to the workplace Wi-Fi network and you’ve just started downloading a large file. As you leave Wi-Fi range your phone automatically switches to the fancy new 5G cellular network. With plain old TCP, the connection would break and cause an interruption. But QUIC is smarter, it uses a “Connection ID” which is more robust to network changes and provides an active Connection Migration feature for clients to switch without interruption.
+
+Finally, TLS (the Transport Layer Security protocol) is already used to protect HTTP/1.1 and HTTP/2.  QUIC, however, has a **deep integration of TLS 1.3**, protecting both HTTP/3 data and QUIC packet metadata such as packet numbers. Using TLS in this way improves end-user privacy and security and makes continued protocol evolution easier. Combining the transport and cryptographic handshakes means that connection setup takes just a single RTT, compared to TCP’s minimum of 2 and worst case of 4. In some cases, QUIC can even go one step further and send HTTP data along with its very first message, which is called “**0-RTT**”. These fast connection setup times are expected to really help HTTP/3 outperform HTTP/2. We discuss this in more detail below.
+
+#### So will HTTP/3 help?
+
+In conclusion, on the surface HTTP/3 is not really all that different from HTTP/2. It doesn’t really add major new features, but mainly changes how the existing ones work under the surface. The real improvements come from QUIC, which offers faster connection setups, increased robustness and resilience to packet loss. As such, HTTP/3 is expected to do better than HTTP/2 on worse networks, while offering very similar performance on faster systems. However, that is if we can get HTTP/3 working, which can be challenging in practice.
+
+### Deploying and discovering QUIC and HTTP/3
+
+Since QUIC and HTTP/3 run over UDP things aren't as simple as with HTTP/1.1 or HTTP/2. Typically an HTTP/3 client has to first discover that QUIC is available at the server. The recommended method for this is HTTP Alternative Services, as described earlier. On its first visit to a website, a client connects to a server using TCP. It then discovers via Alt-Svc that H3 is available, and can set up a new QUIC connection. The Alt-Svc entry can be cached, allowing subsequent visits to avoid the TCP step, but the entry will eventually become stale and need revalidation. This likely will have to be done for each domain separately, which will probably lead to most page loads using a mix of H1, H2 and H3.
```

Comment:
```suggestion
Since QUIC and HTTP/3 run over UDP things aren't as simple as with HTTP/1.1 or HTTP/2. Typically an HTTP/3 client has to first discover that QUIC is available at the server. The recommended method for this is HTTP Alternative Services, as described earlier. On its first visit to a website, a client connects to a server using TCP. It then discovers via Alt-Svc that HTTP/3 is available, and can set up a new QUIC connection. The Alt-Svc entry can be cached, allowing subsequent visits to avoid the TCP step, but the entry will eventually become stale and need revalidation. This likely will have to be done for each domain separately, which will probably lead to most page loads using a mix of HTTP/1, HTTP/2 and HTTP/3.
```